src/components/checkbox/checkbox.pw.tsx

```diff
@@ -0,0 +1,1052 @@
+      await expect(focusedElement).toHaveCSS(
+        "box-shadow",
+        "rgb(255, 188, 25) 0px 0px 0px 3px"
+      );
+    });
+  }
+);
+
+test.describe("should render Checkbox component and check props", () => {
+  test("should render with data-component", async ({ mount, page }) => {
+    await mount(<CheckboxComponent data-component={CHARACTERS.STANDARD} />);
+
+    const checkbox = getComponent(page, CHARACTERS.STANDARD);
+    await expect(checkbox).toHaveAttribute(
+      "data-component",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  test("should render with data-element", async ({ mount, page }) => {
+    await mount(<CheckboxComponent data-element={CHARACTERS.STANDARD} />);
+
+    const checkbox = checkboxComponent(page);
+    await expect(checkbox).toHaveAttribute("data-element", CHARACTERS.STANDARD);
+  });
+
+  test("should render with data-role", async ({ mount, page }) => {
+    await mount(<CheckboxComponent data-role={CHARACTERS.STANDARD} />);
+
+    const checkbox = checkboxComponent(page);
+    await expect(checkbox).toHaveAttribute("data-role", CHARACTERS.STANDARD);
+  });
+
+  boolVals.forEach((booleanValue) => {
+    test(`should render with checked state set to ${booleanValue}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<CheckboxComponent checked={booleanValue} />);
+
+      const checkboxElement = checkboxRole(page);
+
+      if (boolVals) {
+        expect(checkboxElement).toBeTruthy();
+      } else {
+        expect(checkboxElement).toBeFalsy();
+      }
+    });
+  });
+
+  testData.forEach((label) => {
+    test(`should render with ${label} as a label`, async ({ mount, page }) => {
+      await mount(<CheckboxComponent label={label} />);
+
+      const boxLabel = checkboxLabel(page);
+      await expect(boxLabel).toHaveText(label);
+    });
+  });
+
+  testData.forEach((fieldHelp) => {
+    test(`should render with ${fieldHelp} as fieldHelp`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<CheckboxComponent fieldHelp={fieldHelp} />);
+
+      const preview = fieldHelpPreview(page);
+      await expect(preview).toHaveText(fieldHelp);
+    });
+  });
+
+  test("should render with inline fieldHelp", async ({ mount, page }) => {
+    await mount(
+      <CheckboxComponent fieldHelp="Inline fieldhelp" fieldHelpInline />
+    );
+
+    const inlineFieldHelp = checkboxInlineFieldHelp(page);
+    await expect(inlineFieldHelp).toHaveText("Inline fieldhelp");
+  });
+
+  test("should render with helpAriaLabel", async ({ mount, page }) => {
+    await mount(
+      <CheckboxComponent
+        label="Label For CheckBox"
+        labelHelp="Label Help"
+        helpAriaLabel="This text provides more information for the label"
+      />
+    );
+
+    const checkboxIconElement = checkboxIcon(page);
+    await checkboxIconElement.hover();
+    const checkbooxHelpElement = await checkboxHelpIcon(page);
+    await expect(checkbooxHelpElement).toHaveAttribute(
```

Comment:
Amend to `checkboxHelpElement` from `checkbooxHelpElement` once the const has been amended.